Source/Tools/LinuxEmulation/LinuxSyscalls/FileManagement.cpp

```diff
@@ -356,41 +356,37 @@ FileManager::~FileManager() {
 }
 
 fextl::string FileManager::GetEmulatedPath(const char* pathname, bool FollowSymlink) {
-  if (!pathname ||                  // If no pathname
-      pathname[0] != '/' ||         // If relative
-      strcmp(pathname, "/") == 0) { // If we are getting root
-    return {};
-  }
 
-  auto thunkOverlay = ThunkOverlays.find(pathname);
-  if (thunkOverlay != ThunkOverlays.end()) {
-    return thunkOverlay->second;
-  }
+  FDPathTmpData TmpFilename;
 
   auto RootFSPath = LDPath();
   if (RootFSPath.empty()) { // If RootFS doesn't exist
     return {};
   }
 
-  fextl::string Path = RootFSPath + pathname;
-  if (FollowSymlink) {
-    char Filename[PATH_MAX];
-    while (FEX::HLE::IsSymlink(AT_FDCWD, Path.c_str())) {
-      auto SymlinkSize = FEX::HLE::GetSymlink(AT_FDCWD, Path.c_str(), Filename, PATH_MAX - 1);
-      if (SymlinkSize > 0 && Filename[0] == '/') {
-        Path = RootFSPath;
-        Path += std::string_view(Filename, SymlinkSize);
-      } else {
-        break;
-      }
-    }
+  auto FdPath = GetEmulatedFDPath(AT_FDCWD, pathname, FollowSymlink, TmpFilename);
```

Comment:
Maybe `FdPath` and `Path` shouldn't be declared while their values aren't used :).